packages/@webex/plugin-meetings/src/breakouts/index.ts

```diff
@@ -0,0 +1,206 @@
+    enableBreakoutSession: 'boolean', // appears from the moment you enable breakouts
+    groupId: 'string', // appears from the moment you enable breakouts
+    name: 'string', // only present when in a breakout session
+    sessionId: 'string', // appears from the moment you enable breakouts
+    sessionType: 'string', // appears from the moment you enable breakouts
+    startTime: 'string', // appears once breakouts start
+    status: 'string', // only present when in a breakout session
+    url: 'string', // appears from the moment you enable breakouts
+    locusUrl: 'string', // the current locus url
+  },
+
+  children: {
+    currentBreakoutSession: Breakout,
+  },
+
+  derived: {
+    isInMainSession: {
+      deps: ['sessionType'],
+      /**
+       * Returns true if the user is in the main session
+       * @returns {boolean}
+       */
+      fn() {
+        return this.sessionType === BREAKOUTS.SESSION_TYPES.MAIN;
+      }
+    },
+  },
+
+  /**
+   * initialize for the breakouts
+   * @returns {void}
+   */
+  initialize() {
+    this.listenTo(this, 'change:status', () => {
+      if (this.status === BREAKOUTS.STATUS.CLOSING) {
+        this.trigger(BREAKOUTS.EVENTS.BREAKOUTS_CLOSING);
+      }
+    });
+    this.debouncedQueryRosters = debounce(this.queryRosters, 10, {
+      leading: true,
+      trailing: false
+    })
+    this.listenTo(this.breakouts, 'add', () => {
+      this.debouncedQueryRosters();
+    });
+    this.listenToBroadcastMessages();
+    this.listenToBreakoutRosters();
+  },
+
+  /**
+   * Calls this to clean up listeners
+   */
+  cleanUp() {
+    this.stopListening()
+  },
+
+  /**
+   * Update the current locus url of the meeting
+   */
+  locusUrlUpdate(locusUrl) {
+    this.set('locusUrl', locusUrl);
+  },
+
+  /**
+   * The initial roster lists need to be quried because you don't
+   * get a breakout.roster event when you join the meeting
+   */
+  queryRosters() {
+    this.webex.request(
+      {uri: `${this.url}/roster`, qs: {locusUrl: btoa(this.locusUrl)}
+    }).then((result) => {
+      const {body: {rosters}} = result;
+      
+      rosters.forEach(({locus}) => {
+        this.handleRosterUpdate(locus);
+      })
+
+      this.trigger(BREAKOUTS.EVENTS.MEMBERS_UPDATE);
+    })
+  },
+
+  handleRosterUpdate(locus) {
+    const sessionId = locus.controls?.breakout?.sessionId;
+
+    const session = this.breakouts.get(sessionId);
+
+    if (!session) {
+      return;
+    }
+
+    session.parseRoster(locus);
+  },
+
+  /**
+   * Sets up listener for broadcast messages sent to the breakout session
+   */
+  listenToBroadcastMessages() {
+    this.listenTo(this.webex.internal.llm, 'event:breakout.message', (event) => {
+      const {data: {senderUserId, sentTime, message}} = event;
+
+      this.trigger(BREAKOUTS.EVENTS.MESSAGE, {
+        senderUserId,
+        sentTime,
+        message,
+        // FIXME: This is only the current sessionId
```

Comment:
is this because the new llm logic you added initially automatically connects to the new dataChannelURL and this could happen half way through this action (receiving the `event:breakout.message` and triggering it further)?

Reply:
Yes. I am happy to leave this edge case at this stage though. It would only be a problem in cases where the application developer was filtering the broadcast messages in their app by sessionId and dropped a message because it wasn't for the current room.

Reply:
yeah, don't mind it here, was just trying to understand why was needed.